examples/sphere/shallow_water.jl

```diff
@@ -66,132 +104,61 @@ end
-    const ϕ₂ = 45.0
-    const eₙ = exp(-4.0 / (deg2rad(ϕ₁) - deg2rad(ϕ₀))^2)
-else # default case, steady-state test case
-    const u0 = 2 * pi * R / (12 * 86400)
-    const h0 = 2.94e4 / g
-end
-
-# Set up discretization
-ne = 9 # the rossby_haurwitz test case's initial state has a singularity at the pole. We avoid it by using odd number of elements
-Nq = 4
-
-domain = Domains.SphereDomain(R)
-mesh = Meshes.EquiangularCubedSphere(domain, ne)
-quad = Spaces.Quadratures.GLL{Nq}()
-if usempi
-    grid_topology = Topologies.DistributedTopology2D(context, mesh)
-    global_grid_topology = Topologies.Topology2D(mesh)
-    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
-    global_space = Spaces.SpectralElementSpace2D(global_grid_topology, quad)
-else
-    grid_topology = Topologies.Topology2D(mesh)
-    global_space = space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+Base.@kwdef struct BarotropicInstabilityTest{FT, P} <: AbstractTest
+    params::P = Params{FT}()
+    u_max::FT = 80.0
+    xₑ::FT = 0.3
+    αₚ::FT = 19.09859
+    βₚ::FT = 3.81971
+    h0::FT = 10158.18617 # value for initial height from Tempest https://github.com/paullric/tempestmodel/blob/master/test/shallowwater_sphere/BarotropicInstabilityTest.cpp#L86
+    h_hat::FT = 120.0
+    ϕ₀::FT = 25.71428
+    ϕ₁::FT = 64.28571
+    ϕ₂::FT = 45.0
+    eₙ::FT = exp(-4.0 / (deg2rad(ϕ₁) - deg2rad(ϕ₀))^2)
+    α::FT
 end
-coords = Fields.coordinate_field(space)
+BarotropicInstabilityTest(α::FT) where {FT} =
+    BarotropicInstabilityTest{FT, Params{FT}}(; α = α)
 
 # Definition of Coriolis parameter
-if test_name == rossby_haurwitz_test_name
-    f = map(Fields.local_geometry_field(space)) do local_geometry
-        coord = local_geometry.coordinates
-
-        ϕ = coord.lat
-        λ = coord.long
-
-        f = 2 * Ω * sind(ϕ)
-
-        # Technically this should be a WVector, but since we are only in a 2D space,
-        # WVector, Contravariant3Vector, Covariant3Vector are all equivalent.
-        # This _won't_ be true in 3D however!
-        Geometry.Contravariant3Vector(f)
-    end
-else # all other test cases share the same Coriolis parameter
-    f = map(Fields.local_geometry_field(space)) do local_geometry
-        coord = local_geometry.coordinates
-        ϕ = coord.lat
-        λ = coord.long
-
-        f = 2 * Ω * (-cosd(λ) * cosd(ϕ) * sind(α) + sind(ϕ) * cosd(α))
-
+function set_coriolis_parameter(space, test::AbstractTest)
+    Ω = test.params.Ω
+    α = test.α
+    f_rossby_haurwitz(Ω, ϕ, λ, α) = 2 * Ω * sind(ϕ) # for rossby_haurwitz test
+    f_default(Ω, ϕ, λ, α) = # all other test cases
+        2 * Ω * (-cosd(λ) * cosd(ϕ) * sind(α) + sind(ϕ) * cosd(α))
+    f_coriolis = test isa RossbyHaurwitzTest ? f_rossby_haurwitz : f_default
+
+    return map(Fields.local_geometry_field(space)) do local_geometry
+        ϕ = local_geometry.coordinates.lat
+        λ = local_geometry.coordinates.long
+        f = f_coriolis(Ω, ϕ, λ, α) #2 * Ω * sind(ϕ)
```

Comment:
```suggestion
        f = f_coriolis(Ω, ϕ, λ, α)
```
since the comment only applies to the special case